EngineLayer/FdrAnalysis/PEPValueAnalysisGeneric.cs

```diff
@@ -135,63 +142,146 @@ public static string ComputePEPValuesForAllPSMsGeneric(List<PeptideSpectralMatch
-            foreach (PeptideSpectralMatch psm in psms)
-            {
-                var ss = psm.BestMatchingPeptides.Select(b => b.Peptide.FullSequence).ToList();
-                sequences.Add(String.Join("|", ss));
-            }
+            List<string> filenames = psms.Select(f => f.FullFilePath).ToList();
 
-            var s = sequences.GroupBy(i => i);
+            filenames = filenames.Distinct().ToList();
 
-            foreach (var grp in s)
+            foreach (string filename in filenames)
             {
-                sequenceToPsmCount.Add(grp.Key, grp.Count());
-            }
-            return sequenceToPsmCount;
-        }
+                Dictionary<int, List<double>> hydrobophobicites = new Dictionary<int, List<double>>();
+                Dictionary<int, Tuple<double, double>> averagesCommaStandardDeviations = new Dictionary<int, Tuple<double, double>>();
 
-        /// <summary>
-        /// This method goes throught the complete set of peptide spectral matches and counts the number of appearances for each accession number.
-        /// This includes all of the ambiguous entries. So, if one peptide spectral match has three peptides with three accessions, all are counted.
-        /// </summary>
-        /// <param name="psms"></param>
-        /// <returns></returns>
-        public static Dictionary<string, int> GetAccessionCounts(List<PeptideSpectralMatch> psms)
-        {
-            Dictionary<string, int> accessionCountDictionary = new Dictionary<string, int>();
-
-            foreach (PeptideSpectralMatch psm in psms)
-            {
-                if (psm != null)
+                foreach (PeptideSpectralMatch psm in psms.Where(f => (f.FullFilePath == filename || f.FullFilePath == null) && f.FdrInfo.QValue <= 0.01))
                 {
-                    foreach (var (Notch, Peptide) in psm.BestMatchingPeptides)
+                    foreach ((int notch, PeptideWithSetModifications pwsm) in psm.BestMatchingPeptides)
                     {
-                        string accession = Peptide.Protein.Accession;
-                        if (accession != null)
+                        if (pwsm.BaseSequence.Equals(pwsm.FullSequence) && !modified)
+                        {
+                            double predictedHydrophobicity = calc.ScoreSequence(pwsm);
+                            int possibleKey = (int)Math.Round(psm.ScanRetentionTime, 0);
+                            if (hydrobophobicites.ContainsKey(possibleKey))
+                            {
+                                hydrobophobicites[possibleKey].Add(predictedHydrophobicity);
+                            }
+                            else
+                            {
+                                hydrobophobicites.Add(possibleKey, new List<double>() { predictedHydrophobicity });
+                            }
+                        }
+                        else if (!pwsm.BaseSequence.Equals(pwsm.FullSequence) && modified)
```

Comment:
you don't need to change this, but it would be faster to do `peptide.AllModsOneIsNterminus.Any()` instead of doing a string comparison